libpod/container.go

```diff
@@ -833,22 +826,24 @@ func (c *Container) CleanupStorage() error {
 	return c.cleanupStorage()
 }
 
+// cleanupStorage unmounts and cleans up the container's root filesystem
 func (c *Container) cleanupStorage() error {
+	if !c.state.Mounted {
+		// Already unmounted, do nothing
+		return nil
+	}
 
-	if c.state.Mounted {
-		for _, mount := range c.config.Mounts {
-			if err := unix.Unmount(mount, unix.MNT_DETACH); err != nil {
-				if err != syscall.EINVAL {
-					logrus.Warnf("container %s failed to unmount %s : %v", c.ID(), mount, err)
-				}
+	for _, mount := range c.config.Mounts {
+		if err := unix.Unmount(mount, unix.MNT_DETACH); err != nil {
+			if err != syscall.EINVAL {
+				logrus.Warnf("container %s failed to unmount %s : %v", c.ID(), mount, err)
 			}
 		}
-		c.config.Mounts = []string{}
+	}
```

Comment:
You did not clear out c.config.Mounts?  This could cause the caller to make bad assumptions.

Reply:
Won't be saved in the database regardless - `c.config` will never change in the DB. As it is, we'll just rely on IsMounted to see if the requested mounts are set up